nemo/tron/converter/common.py

```diff
@@ -0,0 +1,346 @@
+
+    def to(self, dtype):
+        # pylint: disable=C0115,C0116
+        for k, v in self._state_dict.items():
+            if v.dtype != dtype:
+                logger.warning(f"Converting {k} from {v.dtype} (source model) to {dtype} (target model)")
+            self._state_dict[k] = v.to(dtype)
+
+
+class BaseImporter:
+    def __init__(self, input_path: str | Path, output_path: str | Path):
+        self.input_path = Path(input_path) if isinstance(input_path, str) else input_path
+        self.output_path = Path(output_path) if isinstance(output_path, str) else output_path
+        (self.output_path / HF_ASSETS_DIR).mkdir(parents=True, exist_ok=True)
+        self._hf_config = None
+        self._tron_config = None
+        self.hf_config.save_pretrained(str(self.output_path / HF_ASSETS_DIR))
+
+    @property
+    def tokenizer(self) -> "_HuggingFaceTokenizer":
+        """Get the tokenizer for the HF model.
+
+        Returns:
+            _HuggingFaceTokenizer: Tokenizer instance initialized from the HF model's tokenizer
+        """
+
+        return _HuggingFaceTokenizer(
+            save_hf_tokenizer_assets(str(self.input_path), str(self.output_path / HF_ASSETS_DIR))
+        )
+
+    def init_tron_model(self, cfg: GPTConfig | T5Config):
+        with megatron_cpu_init_context(cfg):
+            model = cfg.configure_model(tokenizer=self.tokenizer)
+        return [model]
+
+    def init_hf_model(self):
+        raise NotImplementedError
+
+    def convert_state(self, source, target):
+        raise NotImplementedError
+
+    @property
+    def hf_config(self) -> "PretrainedConfig":
+        raise NotImplementedError
+
+    @property
+    def tron_config(self) -> GPTConfig | T5Config:
+        raise NotImplementedError
+
+    def apply(self) -> Path:
+        """Run the conversion from HF to NeMo format.
+
+        Args:
+            output_path: Path where the converted model will be saved
+
+        Returns:
+            Path: Path to the saved NeMo model
+        """
+        source = self.init_hf_model()
+
+        with temporary_distributed_context():
+            target = self.init_tron_model(self.tron_config)
+            self.convert_state(source, target[0])
+            state = GlobalState()
+            state.cfg = ConfigContainer(
+                model_config=self.tron_config,
+                train_config=None,
+                optimizer_config=OptimizerConfig(use_distributed_optimizer=False),
+                ddp_config=None,
+                scheduler_config=None,
+                dataset_config=None,
+                logger_config=LoggerConfig(),
+                tokenizer_config=None,
+                checkpoint_config=CheckpointConfig(
+                    async_save=False, save=str(self.output_path), save_optim=False, ckpt_format="torch_dist"
+                ),
+                dist_config=None,
+                ft_config=None,
+                straggler_config=None,
+                profiling_config=None,
```

Comment:
these are None by default - why explicitly include them this way?

Reply:
Good catch, will remove